test/repr/main.ml

```diff
@@ -20,11 +20,15 @@ let of_bin_string t = T.unstage (T.of_bin_string t)
 let encode_bin t = T.unstage (T.encode_bin t)
 let decode_bin t = T.unstage (T.decode_bin t)
 let size_of t = T.unstage (T.size_of t)
+let sub_bytes len off byt = if len = off then byt else Bytes.sub byt 0 off
+let sub_string len off byt = Bytes.to_string @@ sub_bytes len off byt
 
-let with_buf f =
-  let buf = Buffer.create 10 in
-  f (Buffer.add_string buf);
-  Buffer.contents buf
+let with_byt ty v size_of f =
```

Comment:
nitpicking, but do you mind renaming the function into `with_bytes` and `ty` to `t` for consistency